creusot/src/backend/program.rs

```diff
@@ -622,6 +622,42 @@ impl<'tcx> Place<'tcx> {
     }
 }
 
+pub(crate) fn borrow_generated_id<V: std::fmt::Debug, T: std::fmt::Debug>(
+    original_borrow: Exp,
+    projection: &[ProjectionElem<V, T>],
+) -> Exp {
+    let mut borrow_id = Exp::Call(
+        Box::new(Exp::pure_qvar(QName::from_string("Borrow.get_id").unwrap())),
+        vec![original_borrow], // TODO: should be "nested" (how ?)
+    );
+    // NOTE: no Deref should be present.
+    for proj in projection {
+        match proj {
+            ProjectionElem::Deref => {
+                unreachable!(
+                    "the final borrow analysis should have garanteed that only one `Deref` appears"
+                )
+            }
```

Comment:
This is still possible if this is the Deref of a Box

Reply:
Indeed, I changed some things in `NotFinalPlaces::is_final_at` to handles boxes, let me know if this is satisfactory